src/library/basetracktablemodel.cpp

```diff
@@ -159,6 +123,21 @@ void BaseTrackTableModel::initTableColumnsAndHeaderProperties(
     m_columnCache.setColumns(tableColumns);
     if (m_columnHeaders.size() < tableColumns.size()) {
         m_columnHeaders.resize(tableColumns.size());
+        // Init the mapping of all columns, even for internal columns that are
+        // hidden/invisible. Otherwise mapColumn() would not return a valid result
+        // for those columns.
+        for (auto columnValue = 0; columnValue < ColumnCache::NUM_COLUMNS; ++columnValue) {
+            const auto column = static_cast<ColumnCache::Column>(columnValue);
+            DEBUG_ASSERT(column != ColumnCache::COLUMN_LIBRARYTABLE_INVALID);
+            const auto headerIndex = m_columnCache.fieldIndex(column);
+            if (headerIndex < 0) {
+                // Missing table column.
+                continue;
+            }
+            DEBUG_ASSERT(headerIndex < m_columnHeaders.size());
+            m_columnHeaders[headerIndex].column = column;
```

Comment:
Out of scope of this PR.

IMHO the negligible performance overhead for initialitzing the mapping of some columns twice is not worth the maintenance burden.

Reply:
At least this is redundant now: 
https://github.com/mixxxdj/mixxx/blob/dc8fba5806cc086b628410eae4d0111261e30758/src/library/basetracktablemodel.cpp#L259-L262

Reply:
No. Because `setHeaderProperties()` is protected and might be called from another context by derived classes. Not only from within `initTableColumnsAndHeaderProperties()`.

I am not able to make a qualified decision here.

Reply:
You have decided to introduce the code here which introduces the first go of redundnandeny: 
https://github.com/mixxxdj/mixxx/blob/dc8fba5806cc086b628410eae4d0111261e30758/src/library/basetracktablemodel.cpp#L137
Can this than be removed?

Reply:
> Because `setHeaderProperties()` is protected ...

It looks like the protected state is wrong and it can be made private.

Reply:
> You have decided to introduce the code here which introduces the first go of redundnandeny:
> 
> https://github.com/mixxxdj/mixxx/blob/dc8fba5806cc086b628410eae4d0111261e30758/src/library/basetracktablemodel.cpp#L137
> 
> Can this than be removed?

This line of code is the essence of this PR.

Reply:
> > Because `setHeaderProperties()` is protected ...
> 
> It looks like the protected state is wrong and it can be made private.

Its is used by `BaseSqlTableModel::initHeaderProperties()`. Within the same context, but this is could not be guaranteed for other derived classes.